src/ASM/ASMs2DLag.C

```diff
@@ -773,3 +773,45 @@ bool ASMs2DLag::write(std::ostream& os, int) const
 {
   return this->writeLagBasis(os, "quad");
 }
+
+
+int ASMs2DLag::findElement(double u, double v,
+                           double* xi, double* eta) const
+{
+  double du = 1.0 / (nx-1);
+  double dv = 1.0 / (ny-1);
+
+  int elmx = std::min(nx-2.0, floor(u / du));
+  int elmy = std::min(ny-2.0, floor(v / dv));
+
+  if (xi)
+    *xi   = -1.0 + (u - elmx*du)*2.0 / du;
+  if (eta)
+    *eta  = -1.0 + (v - elmy*dv)*2.0 / dv;
+
+  return 1 + elmx + elmy*(nx-1);
+}
+
+
+bool ASMs2DLag::evaluate (const ASMbase* basis, const Vector& locVec,
+                          RealArray& vec, int basisNum) const
```

Comment:
This method is not declared in the header file (at least not in this commit).
But now I see it is coming in next commit, so please move this implementation also there to keep the history consistent.

Reply:
yeah, screwed up the splitting here. remedied.